gtsfm/utils/geometry_comparisons.py

```diff
@@ -192,3 +194,51 @@ def compare_global_poses(
             for (wTi, wTi_) in zip(wTi_list, wTi_list_)
         ]
     )
+
+
+def compute_relative_rotation_angle(
+    R_1: Optional[Rot3], R_2: Optional[Rot3]
+) -> float:
+    """Compute the angle between two rotations.
+
+    Note: the angle is the norm of the angle-axis representation.
+
+    Args:
+        R_1: the first rotation.
+        R_2: the second rotation.
+
+    Returns:
+        the angle between two rotations.
+    """
+
+    if R_1 is None or R_2 is None:
+        return np.pi
+
+    relative_rot = R_1.between(R_2)
+
+    origin = Rot3()
+
+    relative_rot_axisangle = origin.Logmap(relative_rot)
+    relative_rot_angle = np.linalg.norm(relative_rot_axisangle) % (2 * np.pi)
+
+    return relative_rot_angle
+
+
+def compute_relative_unit_translation_angle(
+    U_1: Optional[Unit3], U_2: Optional[Unit3]
+) -> float:
+    """Compute the angle between two unit-translations.
+
+    Args:
+        U_1: the first unit-translation.
+        U_2: the second unit-translation.
+
+    Returns:
+        the angle between the two unit-vectors.
+    """
+    if U_1 is None or U_2 is None:
+        return np.pi
+
+    dot_product = np.dot(U_1.point3(), U_2.point3())
```

Comment:
Doesnt Unit3 have a dot() method?

Reply:
Oh it does. It is not exposed in python yet. I'll maybe leave this as a TODO.